discovery/gossiper.go

```diff
@@ -426,23 +427,30 @@ func (d *AuthenticatedGossiper) Start() error {
 }
 
 // Stop signals any active goroutines for a graceful closure.
-func (d *AuthenticatedGossiper) Stop() {
+func (d *AuthenticatedGossiper) Stop() error {
 	if !atomic.CompareAndSwapUint32(&d.stopped, 0, 1) {
-		return
+		return nil
 	}
 
 	log.Info("Authenticated Gossiper is stopping")
 
 	d.blockEpochs.Cancel()
 
 	d.syncerMtx.RLock()
+	var errStrings []string
```

Comment:
I think it's enough to log any error, an then just return the last encountered one.